pandas/core/arrays/masked.py

```diff
@@ -299,6 +301,16 @@ def take(
 
         return type(self)(result, mask, copy=False)
 
+    def isin(self, values) -> "BooleanArray":
+
+        from pandas.core.arrays import BooleanArray
```

Comment:
@jbrockmendel Hi, I copied some of your codes from https://github.com/pandas-dev/pandas/pull/38422. Would like to explain for me why do we have to import the `BooleanArray` inside `isin` method? Not quite understand that, thanks.

Reply:
BooleanArray is a subclass of BaseMaskedArray, so importing BooleanArray at the top of this file would be a circular import